datafusion-cli/src/object_storage.rs

```diff
@@ -39,17 +39,26 @@ pub async fn get_s3_object_store_builder(
     url: &Url,
     aws_options: &AwsOptions,
 ) -> Result<AmazonS3Builder> {
+    let AwsOptions {
+        access_key_id,
+        secret_access_key,
+        session_token,
+        region,
+        endpoint,
```

Comment:
One bug was that `endpoint ` was not read here so I changes this to destructure the object so the compiler can check we don't forget fields